src/daemon/control.rs

```diff
@@ -726,10 +740,212 @@ pub fn watchtowers_status(revaultd: &RevaultD) -> Vec<ServerStatus> {
             });
         }
     }
-
     watchtowers
 }
 
+// FIXME: probably this function should be in revault_tx, not here...
+fn build_cpfp_tx(
+    to_be_cpfped: &Txid,
+    feerate: u64,
+    listunspent: Vec<ListUnspentResult>,
+    revaultd: &RevaultD,
+) -> Result<PartiallySignedTransaction, RpcControlError> {
+    let cpfp_spend_weight = revaultd
+        .cpfp_descriptor
+        .inner()
+        .max_satisfaction_weight()
+        .unwrap();
+    // Calculating the effective amount of the inputs
+    let mut listunspent: Vec<(ListUnspentResult, Amount)> = listunspent
+        .into_iter()
+        .map(|u| {
+            let effective_amount = SignedAmount::from_sat(u.txo.value as i64)
+                - SignedAmount::from_sat(feerate as i64 * cpfp_spend_weight as i64);
+            (u, effective_amount)
+        })
+        .filter_map(|(u, effective_amount)| {
+            // We don't want to add uneconomical inputs!
+            if let Ok(unsigned) = effective_amount.to_unsigned() {
+                Some((u, unsigned))
+            } else {
+                None
+            }
+        })
+        .collect();
+
+    // This will sort the vector in ascending order. Since
+    // we're going to pop() from it, we're using a largest
+    // first CS.
+    listunspent.sort_unstable_by_key(|u| u.1);
+
+    // Let's pick the input to be cpfped
+    let to_be_cpfped_index = listunspent
+        .iter()
+        .position(|l| &l.0.outpoint.txid == to_be_cpfped)
+        .ok_or(RpcControlError::MissingCpfpInput)?;
+
+    // Pushing it at the end of the vector, so it gets picked first
+    let u = listunspent.remove(to_be_cpfped_index);
+    listunspent.push(u);
+    let mut inputs_sum = Amount::from_sat(0);
+
+    let transaction = BitcoinTransaction {
+        version: 2,
+        lock_time: 0,
+        input: vec![],
+        output: vec![TxOut {
+            // FIXME: Maybe using an OP_RETURN for the whole CS and
+            // then swapping to P2WSH (in case of change) will cause
+            // troubles with fee calculation?
+            value: 0,
+            script_pubkey: Script::new_op_return(&[]),
+        }],
+    };
+    let mut psbt =
+        PartiallySignedTransaction::from_unsigned_tx(transaction).expect("We don't sign the tx");
+
+    loop {
+        // TODO: I should consider the package fees and not the individual transaction fees
+        // to be more precise
```

Comment:
More than that, it's actually a requirement for this to make sense at all :sweat_smile:

Reply:
Same as above, it's in my todo list (see the PR description) with high prio, but you can probably look at the general structure of the cpfp (the tracking etc) without this detail. That's mostly why I opened a PR with a dummy, full of TODOs CPFP.